courses-rest/src/test/java/no/sikt/nva/fs/CoursesByInstitutionOfLoggedInUserHandlerTest.java

```diff
@@ -1,65 +1,277 @@
+        assertThat(gatewayResponse.getBodyObject(Course[].class), emptyArray());
+        assertThat(appender.getMessages(), containsString(LOG_MESSAGE_PREFIX_FS_COMMUNICATION_PROBLEM + "215"));
+    }
+    
     @Test
-    void dummyTest() throws IOException {
+    void shouldReturnEmptyListOfCoursesAndLogProblemIfFsReturnsNonSuccess() throws IOException {
+        // prepare:
+        final InputStream input = createRequest(SUPPORTED_INSTITUTION_PATH);
+        final TestAppender appender = LogUtils.getTestingAppenderForRootLogger();
+        
+        stubFor(get(urlPathEqualTo(TAUGHT_COURSES_URL_PATH))
+            .willReturn(WireMock.aResponse()
+                .withStatus(HttpURLConnection.HTTP_INTERNAL_ERROR)));
+        
+        this.handler = new CoursesByInstitutionOfLoggedInUserHandler(environment, FIXED_CLOCK);
+        
+        // execute:
+        handler.handleRequest(input, output, context);
+        
+        // verify:
+        final GatewayResponse<Course[]> gatewayResponse = GatewayResponse.fromOutputStream(output, Course[].class);
+        
+        assertEquals(HttpURLConnection.HTTP_OK, gatewayResponse.getStatusCode());
+        assertThat(gatewayResponse.getHeaders(), hasKey(CONTENT_TYPE));
+        assertThat(gatewayResponse.getHeaders(), hasKey(ACCESS_CONTROL_ALLOW_ORIGIN));
+        assertThat(gatewayResponse.getBodyObject(Course[].class), emptyArray());
+        assertThat(appender.getMessages(), containsString(
+            HttpUrlConnectionFsClient.UNEXPECTED_STATUS_CODE_LOG_MESSAGE_PREFIX + "500"));
+    }
+    
+    @Test
+    void shouldReturnEmptyListOfCoursesWhenInstitutionDoesNotHaveFsAccess() throws IOException {
+        // prepare:
         final InputStream input = new HandlerRequestBuilder<Void>(restApiMapper)
-                                      .withNvaUsername(randomString())
-                                      .withCustomerId(randomUri())
-                                      .build();
-
+            .build();
+        
+        this.handler = new CoursesByInstitutionOfLoggedInUserHandler(environment, FIXED_CLOCK);
+        
+        // execute:
         handler.handleRequest(input, output, context);
-
-        final GatewayResponse<CoursesByInstitutionResponse> gatewayResponse =
-            GatewayResponse.fromOutputStream(output, CoursesByInstitutionResponse.class);
-
+        
+        // verify:
+        final GatewayResponse<Course[]> gatewayResponse = GatewayResponse.fromOutputStream(output, Course[].class);
+        
+        assertEquals(HttpURLConnection.HTTP_OK, gatewayResponse.getStatusCode());
+        assertThat(gatewayResponse.getHeaders(), hasKey(CONTENT_TYPE));
+        assertThat(gatewayResponse.getHeaders(), hasKey(ACCESS_CONTROL_ALLOW_ORIGIN));
+        assertThat(gatewayResponse.getBodyObject(Course[].class), emptyArray());
+    }
+    
+    @Test
+    void shouldReturnCoursesWhenInstitutionHasFsAccessAfterSummer() throws IOException {
```

Comment:
the expectation that is stated in natural language does not make apparent what the real expectations are. What I understand is that the institution has access to FS after summer (it did not have access before summer) and it gets some courses back.

Reply:
Improved name of test methods to better reflect expectations